Pull request #1 from teodoroven

teodoroven opened the pull request:
• Added row with total hours in teach plan page
• Added the trimester number parser into the course number + trimester 
number in teach plan page
• Fixed consultations button: now it remembers your choice in timetable 
page

teodoroven committed:
Pull request #1 from teodoroven
• Added row with total hours in teach plan page
• Added the trimester number parser into the course number + trimester 
number in teach plan page
• Fixed consultations button: now it remembers your choice in timetable 
page

VladislavMaksimov commented on scripts.js:
```diff
@@ -639,4 +642,61 @@ function stylePages() {
+function totalHours() {
+	// Только на странице учебного плана
+	if (window.location.href.toLowerCase().includes('teach_plan')) {
+		const tables = document.querySelectorAll('table.common');
```
В целом всё окей, но нужно кое-что исправить.

1. Ты вставляешь строку прямо в таблицу, чтобы её отобразить, браузер внутри каждой таблицы создаёт дополнительное tbody для этой строки. Это чревато неожиданными багами, и один из них в том, что у этой строки не отображается border. Нужно вставлять твою строку внутрь tbody. Если в этом селекторе написать table.common > tbody, по идее, всё должно сработать.

teodoroven replied:
Пофиксил

VladislavMaksimov commented on scripts.js:
```diff
@@ -639,4 +642,61 @@ function stylePages() {
+		headers.forEach(header => {
+			let content = header.textContent; // Получаем текст заголовка
+	    let number = Number(content.slice(0,content.indexOf(' '))); // Достаём из него номер триместра
+			header.textContent += ` (${Math.floor(number / 3)+1} курс, ${(number-1) % 3 + 1} триместр)`; // Вычисляем номер курса и добавляем к заголовку
```
Номер курса в каждом третьем триместре считается неправильно.  Например, 3 / 3 + 1 = 1 + 1 = 2 ===> 3 триместр 1 курса становится 3 триместром 2 курса. 

Я предлагаю просто использовать Math.ceil вместо Math.floor, ну и соответственно убрать из расчёта курса +1.

https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Math/ceil

teodoroven replied:
Пофиксил

VladislavMaksimov commented on scripts.js:
```diff
@@ -639,4 +642,61 @@ function stylePages() {
+		// Добавляем функцию на нажатие для сохранения состояния чекбокса
+		if (button) {
+			button.onclick = () => {
+	      localStorage.setItem('showConsultations',String(!checkbox.checked));
```
Здесь чекбокс прыгает меняет состояние на предыдущее после нажатия. Кажется, надо кидать в локалсторейдж checkbox.checked, а не противоположное ему значение.

teodoroven replied:
Это актуально для onchange, но если использовать onclick, то обработчик срабатывает раньше, чем input меняет состояние checked, поэтому отрицание здесь к месту.

VladislavMaksimov commented on scripts.js:
```diff
@@ -639,4 +642,61 @@ function stylePages() {
+
+// Remember Consultations Value
+function consultationsMemorizer() {
+	const button = document.querySelector('.timetable-btn.consultations');
```
Ну и последнее: добавь, плз, на всякий случай проверку того, что открыта страница с расписанием. Класс .consultations, наверное, действительно есть только в расписании, но это ж ЕТИС :) Он может внезапно вылезти на какой-то левой странице.

teodoroven replied:
Сделал

teodoroven committed:
Исправления:
• Исправлена неисправность с переводом номера триместра в номер курса
• Общее количество часов отображается только если есть что суммировать
• Исправлена неисправность с отсутствием рамки у строки с общим 
количеством часов в таблице
• Изменена логика consultationsMemorizer

teodoroven committed:
Вдогонку:
• consultationsMemorizer теперь работает только на странице с 
расписанием
